### PR TITLE
docs: some improvements to model docs

### PR DESCRIPTION
**Pull request by jonbretman**
* Added section on optional fields (was missing)
* Added some context around how foreign keys work with relationships
* Simpler and more real-world example of a compound unique constraint
* Some more context around how one-to-one relationships work

### Table of contents
| Before   | After                                                                      | 
| ----- | ----------------------------------------------------------------------------- | 
| Screenshot 2023-12-08 at 12 00 10    | Screenshot 2023-12-08 at 12 00 00 |

### CODE REVIEW

File: pages/models.mdx [hunk 1]
```diff
@@ -40,16 +40,51 @@ The basic built-in Keel types are:
 
 As previously mentioned, all models **automatically** have the following fields:
 
-- `id` - a field of type `ID` that can be used to identify an individual record
+- `id` - a field of type `ID` that can be used to identify an individual record (we use ksuid's for ID's)
```
Comment (TejasQ) [inline pages/models.mdx:43]: Possessive or plural?

**Review by TejasQ — APPROVED** [review]
Looks good!

File: pages/models.mdx (hunk 1 above)
Comment (jonbretman) [reply pages/models.mdx:43]: Oh I have no idea - this is my grammar blind spot. It might actually be better to just say "we use KSUID for ID's".

Comment (jonbretman) [reply pages/models.mdx]: Changed to "KSUIDs"

**Pull request merged**